### PR TITLE
Added ArrayLike In Series.clip

### LINKED ISSUE #437: BUG: Series.clip should accept array-like for upper and lower
**Describe the bug**
Series.clip currently types upper and lower arguments as accepting Optional[float]. However they also accept array-like.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
2. Indicate which type checker you are using (`mypy` or  `pyright`).
3. Show the error message received from that type checker while checking your example.

```
ser = pd.Series([1, 2, 3])
ser.clip(upper=ser)
```

With mypy 0.991: `error: Argument "upper" to "clip" of "Series has incompatible type "Series[Any]"; expected "Optional[float]"`

### PR DESCRIPTION
**Pull request by ramvikrams**
- [x] Closes #437 
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

Sir can you please check the tests.

### CODE REVIEW

File: tests/test_series.py [hunk 1]
```diff
@@ -1330,3 +1330,13 @@ def test_logical_operators() -> None:
...
     )
+
+
+def test_ArrayLike_and_clip() -> None:
+    s = pd.Series([1, 2, 3, 4])
+    check(
+        assert_type((s.clip(upper=s)), "pd.Series[Any]"), pd.Series[Any], pd.Series[Any]
+    )
+    check(
+        assert_type((s.clip(lower=s)), "pd.Series[Any]"), pd.Series[Any], pd.Series[Any]
```
Comment (Dr-Irv) [inline tests/test_series.py:1341]: You can do `assert_type(s.clip(lower=s), pd.Series), pd.Series)` here

Also, remove the extra parentheses around the calls to `s.clip()`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_series.py (hunk 1 above)
Comment (ramvikrams) [reply tests/test_series.py:1341]: > You can do `assert_type(s.clip(lower=s), pd.Series), pd.Series)` here

Are you sujjesting `assert_type((s.clip(lower=s), pd.Series), pd.Series)` or `assert_type(s.clip(lower=s), pd.Series)` but in both cases mypy fails

Comment (Dr-Irv) [reply tests/test_series.py:1341]: Use `check` with `assert_type` and use `AnyArrayLike` not `ArrayLike`

You are going to have to start figuring out these things by yourself.   Look at the error messages and debug the code.

Comment (ramvikrams) [reply tests/test_series.py]: Yes Sir, done with the changes didn't add check because it was giving TypeError.

Comment (ramvikrams) [reply tests/test_series.py]: > You are going to have to start figuring out these things by yourself. Look at the error messages and debug the code.

yes sir

File: tests/test_series.py [hunk 2]
```diff
@@ -1330,3 +1330,9 @@ def test_logical_operators() -> None:
         pd.Series,
         bool,
     )
+
+
+def test_AnyArrayLike_and_clip() -> None:
+    ser = pd.Series([1, 2, 3, 4])
+    assert_type(ser.clip(upper=ser), pd.Series)
+    assert_type(ser.clip(lower=ser), pd.Series)
```
Comment (Dr-Irv) [inline tests/test_series.py:1338]: have to do `check`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_series.py (hunk 2 above)
Comment (ramvikrams) [reply tests/test_series.py:1338]: ok sir

Comment (ramvikrams) [reply tests/test_series.py]: sir can you please check the recent commit. the test are wrong I read on internet about TypeError, tried myself but could not fix it

Comment (ramvikrams) [reply tests/test_series.py]: sir can you please check the recent commit. the test are wrong I read on internet about TypeError, tried myself but could not fix it

File: tests/test_series.py [hunk 3]
```diff
@@ -1330,3 +1330,12 @@ def test_logical_operators() -> None:
...
         bool,
     )
+
+
+def test_AnyArrayLike_and_clip() -> None:
+    ser = pd.Series([1, 2, 3])
+    s1=ser.clip(upper=ser)
+    s2=ser.clip(upper=ser)
+    check(assert_type(s1, pd.Series), pd.Series, pd.Series[Any])
+    check(assert_type(s2, pd.Series), pd.Series, pd.Series[Any])
```
Comment (Dr-Irv) [inline tests/test_series.py:1340]: `check(assert_type(s2, pd.Series), pd.Series)`

If you know the type of the object inside the `Series`, then you do `check(assert_type(s2, pd.Series), pd.Series, float)` (for example).  But here we don't know the type, so we just leave out that last argument to `check`. 

Look at the code for `check` to see how it works

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Review by Dr-Irv — APPROVED** [review]
thanks @ramvikrams

**Comment by ramvikrams** [comment]
thanks sir

**Pull request merged**